app/controllers/notification_recipients_controller.rb

```diff
@@ -34,4 +26,27 @@ def find_resource
     super
     @notification_recipient.current_user? || not_found
   end
+
+  def payload
+    Rails.cache.fetch(cache_key, raw: true) do
+      notifications = load_notifications
+
+      {
+        :notifications => notifications,
+        :total => notifications.count
+      }.to_json
+    end
+  end
+
+  def load_notifications
+    NotificationRecipient.
+      where(:user_id => User.current.id, :notification_id => Notification.active).
```

Comment:
Doesn't this cause an extra inner query on a table you (could be) joining in any case?